### PR TITLE
Mark estree and estree-jsx as 1.0

### PR DESCRIPTION
**Pull request by remcohaszing**
Both are type definitions for a specification, not a package. By marking them as version 1.0, people can use “normal” semver ranges to depend on them. Also they are stable, so in that regard it makes sense to mark them as 1.0 too.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: N/A
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

cc @wooorm

### CODE REVIEW

**Review by wooorm — APPROVED** [review]

**Review by wooorm — APPROVED** [review]

**Comment by RReverser** [comment]
> Also they are stable, so in that regard it makes sense to mark them as 1.0 too.

Hm, I'm not sure we can consider them stable given that we're changing / reordering type definitions (like you did just recently). For types with frequent refactorings, where users might import types that will be gone / incompatible in the next version, I think it makes sense to keep 0.x.

**Comment by wooorm** [comment]
They have never really changed, since they were added in 2019, and were updated recently. So, assuming this is all working fine (I believe it is) and doesn’t need changes (\*), it seems very stable to me.

\* Of course, we could change more things, and then it makes sense to do it now, before a major.
Like make more content types extendable. That particularly can go in a minor though as it shouldn’t break things.
Are you thinking of making other changes?

File: types/estree-jsx/tslint.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
     "extends": "@definitelytyped/dtslint/dt.json",
     "rules": {
-        "dt-header": false
+        "dt-header": false,
+        "npm-naming": false
```
[the remark this answers is not in the document]
Comment (sheetalkamat) [reply types/estree-jsx/tslint.json:5]: https://github.com/microsoft/DefinitelyTyped-tools/blob/master/packages/dtslint/docs/npm-naming.md

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/estree-jsx/tslint.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (remcohaszing) [reply types/estree-jsx/tslint.json:5]: Thanks! Much better indeed! Ironically this allows me to reove the disabled rule for `@types/estree`, but not for `@types/estree-jsx`, as the package [`estree-jsx`](https://www.npmjs.com/package/estree-jsx) exists on npm.

**Review by wooorm — APPROVED** [review]

**Review by sheetalkamat — APPROVED** [review]

**Review by wooorm — APPROVED** [review]

**Comment by remcohaszing** [comment]
@RReverser Are you still concerned about the instability of these types? If so, I’ll gladly change the versions to 0.1. Otherwise, I’ll merge this as-is tomorrow.

**Comment by remcohaszing** [comment]
Ready to merge

**Pull request merged**